### PR TITLE
Change caching strategy **Breaking Change**

### PR DESCRIPTION
**Pull request by rubydog**
* Changed `__CACHE__` object type from `list` to `dict`
 * Changed get cache interface: `ContentTypeCache.get(space_id, content_type_id)`

### CODE REVIEW

File: contentful/content_type_cache.py [hunk 1]
```diff
@@ -16,15 +16,15 @@ class ContentTypeCache(object):
     Used for properly coercing Entry fields.
     """
 
-    __CACHE__ = []
+    __CACHE__ = {}
 
     @classmethod
-    def get(cls, content_type_id):
+    def get(cls, space_id, content_type_id):
```
Comment (EricNelson12) [inline contentful/content_type_cache.py:22]: @rubydog Lookin' good!

I think you might need to pass the space_id in [Entry.py:52](https://github.dev/contentful/contentful.py/blob/7105eee74d719ea212370ac62479a1c7a660c87d/contentful/entry.py#L52) :



somehow :thinking:

Comment (rubydog) [reply contentful/content_type_cache.py:22]: @EricNelson12 thank you for pointing that out.

**Review by EricNelson12 — APPROVED** [review]
This is looking good to me :smile: Was thinking it might be good to have a test with two clients with the same content models to make sure the clients pull from the right cache. But maybe that's not necessary!


And just to double check, the below call to `sys` would be different for a different client instance right?

```python
self.sys['space']
```

**Review by ruderngespra — APPROVED** [review]
lgtm 🎉

**Comment by rubydog** [comment]
@EricNelson12 Hmm might make sense to have such a test. Will check and keep you updated.

**Pull request merged**